### PR TITLE
fix: [DEL-5758]: return base64 encoded token when creating token and when UI gets a token

### PR DESCRIPTION
**Pull request by xingchi-jin**
## Describe your changes
To make our installation workflow consistent between raw k8s manifest and other workflows, we need to make UI accessible delegate tokens base64 encoded. Currently the token copied from UI is not encoded, which the token we ask customers to put in the yaml file requires to be base64 encoded.

Test status
Unit tests done
Local tests done
PR ENV done

Test Create Token, Expect token to be base64 encoded
Screen Shot 2023-01-12 at 5 53 43 PM

Token show on the installation commands are base64 encoded
image

Token show in terraform file is base64 encoded
Screen Shot 2023-01-18 at 1 26 56 AM

Launch Docker delegate with a base64 encoded token. Docker delegate runs fine 
Screen Shot 2023-01-18 at 2 45 20 PM
Screen Shot 2023-01-18 at 2 49 59 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)

### CODE REVIEW

File: 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java [hunk 1]
```diff
@@ -64,21 +68,28 @@ public DelegateInstallationCommandServiceImpl(
 
   @Override
   public String getCommand(@NotBlank String commandType, @NotBlank String managerUrl, @NotBlank String accountId) {
-    final Map<String, Boolean> activeStatus =
-        delegateNgTokenService.isDelegateTokenActive(accountId, List.of(DEFAULT_TOKEN_NAME));
-    if (Objects.isNull(activeStatus) || activeStatus.isEmpty()) {
+    final DelegateTokenDetails delegateTokenDetails =
+        delegateNgTokenService.getDelegateToken(accountId, DEFAULT_TOKEN_NAME, true);
+    if (Objects.isNull(delegateTokenDetails)) {
```
Comment (Vbhatia20) [inline 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:73]: is this right if no default token is present, but a custom token is present? what happens in that case

Comment (xingchi-jin) [reply 400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java:73]: in this implementation, it raises exception. 
maybe makes more sense to change it to show "<PUT YOUR TOKEN>" as well. is it ok ?

**Comment by xingchi-jin** [comment]
trigger buildnum

**Comment by raghuAtWings** [comment]
*[`400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java` line 808 at r5](https://reviewable.io/reviews/harness/harness-core/42545#-NLdZoMJ0gKYaWxEFgYP:-NLdZoMK1-v5RLr40MLR:b-ad6e32) ([raw file](https://github.com/harness/harness-core/blob/b270acda693d7067291ea5d699bf83663d2a26bc/400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java#L808)):*
> ```Java
>       @QueryParam("accountId") @NotEmpty String accountId, @QueryParam("orgId") String orgId,
>       @QueryParam("projectId") String projectId, @QueryParam("commandType") @NotEmpty String commandType)
>       throws IOException {
> ```

is the UI sending these params already?

**Review by xingchi-jin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 7 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java` line 808 at r5](https://reviewable.io/reviews/harness/harness-core/42545#-NLdZoMJ0gKYaWxEFgYP:-NLdq1Fr8qNkQ2ZRsZLq:b-pi3jws) ([raw file](https://github.com/harness/harness-core/blob/b270acda693d7067291ea5d699bf83663d2a26bc/400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java#L808)):*
<details><summary><i>Previously, raghuAtWings (Raghvendra Singh) wrote…</i></summary><blockquote>

is the UI sending these params already?
</blockquote></details>

I have informed UI this change.

**Comment by xingchi-jin** [comment]
ping for review

**Review by mdmilic — DISMISSED** [review]
> Reviewed 3 of 5 files at r1, 3 of 4 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 2. unresolved discussions. 

___
*[`400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java` line 86 at r7](https://reviewable.io/reviews/harness/harness-core/42545#-NM-RY1h2KDKbjZJuk1A:-NM-RY1h2KDKbjZJuk1B:bsuovxn) ([raw file](https://github.com/harness/harness-core/blob/e1db709088f99d6869dc1e693f92e947e7efa233/400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java#L86)):*
> ```Java
>     final Map<String, String> values = ImmutableMap.<String, String>builder()
>                                            .put("account_id", accountId)
>                                            .put("token", tokenValue)
> ```

Do we still need token then?
___
*[`400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java` line 808 at r5](https://reviewable.io/reviews/harness/harness-core/42545#-NLdZoMJ0gKYaWxEFgYP:-NM-SSej4LDlFBiOYItU:b4wvmc6) ([raw file](https://github.com/harness/harness-core/blob/b270acda693d7067291ea5d699bf83663d2a26bc/400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java#L808)):*
<details><summary><i>Previously, xingchi-jin (Xingchi Jin) wrote…</i></summary><blockquote>

I have informed UI this change. 
</blockquote></details>

I'm assuming this means we need UI deployment first

**Review by xingchi-jin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 3. unresolved discussions. 

___
*[`400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java` line 86 at r7](https://reviewable.io/reviews/harness/harness-core/42545#-NM-RY1h2KDKbjZJuk1A:-NM1Lokh7arohNYqMxBj:b-jp6hio) ([raw file](https://github.com/harness/harness-core/blob/e1db709088f99d6869dc1e693f92e947e7efa233/400-rest/src/main/java/io/harness/delegate/service/impl/DelegateInstallationCommandServiceImpl.java#L86)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

Do we still need token then?
</blockquote></details>

no. good catch 
___
*[`400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java` line 808 at r5](https://reviewable.io/reviews/harness/harness-core/42545#-NLdZoMJ0gKYaWxEFgYP:-NM1Ldte7_HSNeRzoV_V:bk3zc7w) ([raw file](https://github.com/harness/harness-core/blob/b270acda693d7067291ea5d699bf83663d2a26bc/400-rest/src/main/java/software/wings/resources/DelegateSetupResource.java#L808)):*
<details><summary><i>Previously, mdmilic (Marko Milic) wrote…</i></summary><blockquote>

I'm assuming this means we need UI deployment first
</blockquote></details>

Yes, correct. UI should take the change. I will check agin with @swarajcheguri-harness in today's sync

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 3 of 5 files at r1, 1 of 4 files at r5, 1 of 5 files at r8, 4 of 5 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 2. unresolved discussions.

**Review by mdmilic — APPROVED** [review]
> Reviewed 1 of 5 files at r8, 4 of 5 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 1. unresolved discussions.

**Review by xingchi-jin — COMMENTED** [review]
> Reviewed 3 of 5 files at r1, 1 of 4 files at r5, 1 of 5 files at r8, 4 of 5 files at r9, 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 1. unresolved discussions.

**Review by Vbhatia20 — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42545)* status: 1. unresolved discussions.

**Pull request merged**